swarm/network/stream/messages.go

```diff
@@ -339,7 +339,8 @@ func (p *Peer) handleWantedHashesMsg(ctx context.Context, req *WantedHashesMsg)
 				return fmt.Errorf("handleWantedHashesMsg get data %x: %v", hash, err)
 			}
 			chunk := storage.NewChunk(hash, data)
-			if err := p.Deliver(ctx, chunk, s.priority); err != nil {
+			syncing := true
```

Comment:
i would rather have two deliver methods here too, not a param. lets think

Reply:
Your comment makes sense, but the implementation of those two methods would be exactly the same (at least today). So from the point of view of maintenance, it makes more sense to just use a flag.

Reply:
ok but we can DRY it, i recommend not sending extra info on the wire cos of code issues